packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,648 @@
+    _updateOverlay();
+  }
+
+  // Called when _autocompleteController.textEditingController changes.
+  void _onChangedField() {
+    if (_selection != null) {
+      final String selectionString =
+          _autocompleteController.displayStringForOption(_selection);
+      if (_autocompleteController.textEditingController.text == selectionString) {
+        return;
+      }
+    }
+    setState(() {
+      _selection = null;
+    });
+    _updateOverlay();
+  }
+
+  // Called from buildField when the user submits the field.
+  void _onFieldSubmitted() {
+    final List<T> results = _autocompleteController.results.value;
+    if (results.isEmpty) {
+      return;
+    }
+    _select(results[0]);
+  }
+
+  // Select the given option and update the widget.
+  void _select(T nextSelection) {
+    if (nextSelection == _selection) {
+      return;
+    }
+    setState(() {
+      _selection = nextSelection;
+      final String selectionString =
+          _autocompleteController.displayStringForOption(nextSelection);
+      _autocompleteController.textEditingController.value = TextEditingValue(
+        selection: TextSelection.collapsed(offset: selectionString.length),
+        text: selectionString,
+      );
+      if (widget.onSelected != null) {
+        widget.onSelected(_selection);
+      }
+    });
+  }
+
+  // Hide or show the results overlay, if needed.
+  void _updateOverlay() {
+    if (_shouldShowResults) {
+      assert(_fieldKey.currentContext != null);
+      final RenderBox renderBox = _fieldKey.currentContext.findRenderObject() as RenderBox;
+      _floatingResults?.remove();
+      _floatingResults = OverlayEntry(
+        builder: (BuildContext context) {
+          return _FloatingResults<T>(
+            buildResults: widget.buildResults,
+            fieldSize: renderBox.size,
+            layerLink: _resultsLayerLink,
+            onSelected: _select,
+            results: _autocompleteController.results.value,
+          );
+        },
+      );
+      Overlay.of(context, rootOverlay: true).insert(_floatingResults);
+    } else if (_floatingResults != null) {
+      _floatingResults.remove();
+      _floatingResults = null;
+    }
+  }
+
+  void _listenToController(AutocompleteController<T> autocompleteController) {
+    autocompleteController.results.addListener(_onChangedResults);
+    autocompleteController.textEditingController.addListener(_onChangedField);
+  }
+
+  void _unlistenToController(AutocompleteController<T> autocompleteController) {
+    autocompleteController.results.removeListener(_onChangedResults);
+    autocompleteController.textEditingController.removeListener(_onChangedField);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _autocompleteController = widget.autocompleteController
+        ?? AutocompleteController<T>(options: widget.options);
+    _listenToController(_autocompleteController);
+
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) {
```

Comment:
You have to wait for the layout process to finish to get the current fieldSize, is that the reason this needs to be in a post-frame callback?

Reply:
Yeah that's right.

Reply:
It's sad that we have to wait for the layout to finish. It kinda defeats the purpose of using CompositedTransformX since we need the layout info anyways. I'll create a new issue if there isn't one already.

Reply:
Found one: https://github.com/flutter/flutter/issues/57604